torch_np/_funcs_impl.py

```diff
@@ -913,6 +913,28 @@ def take_along_axis(arr: ArrayLike, indices: ArrayLike, axis):
     return torch.take_along_dim(arr, indices, axis)
 
 
+def put(a: ArrayLike, ind: Sequence[ArrayLike], v: ArrayLike, mode="raise"):
+    if mode != "raise":
+        raise NotImplementedError(f"{mode=}")
+
+    index = torch.concat(ind)
+    index[index < 0] += a.numel()  # normalise negative indices
+    index_u, index_c = torch.unique(index, return_counts=True)
+    duplicated_indices = index_u[index_c > 1]
+    if duplicated_indices.numel() > 0:
+        raise NotImplementedError(
+            "duplicated indices are not supported. duplicated indices: "
+            f"{list(duplicated_indices)}"
+        )
```

Comment:
Data-dependent flow is not great for compilers, so we try to avoid it when it's just used to throw an error.